ui/app/components/feedback/FeedbackTableModal.tsx

```diff
@@ -0,0 +1,74 @@
+import type { FeedbackRow } from "~/utils/clickhouse/feedback";
+import {
+  PageLayout,
+  PageHeader,
+  SectionLayout,
+  SectionsGroup,
+} from "~/components/layout/PageLayout";
+import {
+  SnippetLayout,
+  SnippetContent,
+} from "~/components/layout/SnippetLayout";
+import { CodeMessage, TextMessage } from "~/components/layout/SnippetContent";
+
+interface FeedbackTableModalProps {
+  feedback: FeedbackRow;
+}
+
+export function CommentModal({ feedback }: FeedbackTableModalProps) {
+  if (feedback.type !== "comment" || typeof feedback.value !== "string") {
+    return <div>Invalid comment feedback</div>;
+  }
+
+  return (
+    <PageLayout>
+      <PageHeader label="Comment" name={feedback.id} />
+
+      <SectionsGroup>
+        <SectionLayout>
+          <SnippetLayout>
+            <SnippetContent maxHeight="Content">
+              <TextMessage content={feedback.value} />
+            </SnippetContent>
+          </SnippetLayout>
+        </SectionLayout>
+      </SectionsGroup>
+    </PageLayout>
+  );
+}
+
+export function DemonstrationModal({ feedback }: FeedbackTableModalProps) {
+  if (feedback.type !== "demonstration" || typeof feedback.value !== "string") {
+    return <div>Invalid demonstration feedback</div>;
+  }
+
+  return (
```

Comment:
![image](https://github.com/user-attachments/assets/28b5a887-f7d8-43d7-b44c-8498d21403c5)
one thing I don't love here is that the content horizontally scrolls and it's hard to read for JSON output which is the common case. Can we do something where we see if the text block parses as JSON and then if it's JSON just pretty print it? I think this could probably be used widely in our codebase.

Reply:
if you want to do this in a separate PR then please open an issue!